samples/bookinfo/src/build-services.sh

```diff
@@ -16,18 +16,24 @@
 
 set -o errexit
 
+VERSION=$1
 SCRIPTDIR=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
 ISTIO_VERSION=$(cat $SCRIPTDIR/../../../istio.RELEASE)
 
+if [ "$VERSION" != "$ISTIO_VERSION" ]; then
+    echo You are setting version $VERSION, \
+	 note that it is different from the current Istio version \($ISTIO_VERSION\)
```

Comment:
This VERSION value really has nothing to do with ISTIO_VERSION, since the sample should work with any version of Istio. It's really the "sample version". I think we previously used the version of istio that was being used when the sample was updated, for convenience. It looks like you are proposing to require an explicit version value to be passed in as an arg to the scripts now. What kind of value are you expecting the user to pass in, and shouldn't there be a default? 

Unless the "default" value is ISTIO_VERSION, I don't think we need the above message. If not, maybe it should still have a default. How about 1 greater than the current images used in the yaml file and start using values something like "sample-v1", "sample-v1", etc.?

Reply:
@frankbu is right.. tracking istio version has been a pain. 

But Frank, i think what he is doing is codifying the current convention. The process we followed so far is that when we bump the bookinfo versions, we bump to the same version as the next Istio release.

Reply:
Sticking with the current version is fine with me if we don't want to change anything, but it is a little confusing to tag the sample images with an Istio version, even though it works with any Istio version. It would seem better if we made it clear that the tag represents a version of the sample, really nothing to do with the Istio version that you are running it on.

Reply:
Hmm.. Okay, I agree. It might even be better to break from the istio versions..